src/chain.ts

```diff
@@ -118,7 +119,10 @@ export const chain = (
             if (isPlain) {
                 sourceSocket.end();
             } else {
-                sourceSocket.end(createHttpResponse(502, ''));
+                const unauthorized = [401, /* 403, */ 407];
+                const status = unauthorized.includes(response.statusCode!) ? 597 : 590;
```

Comment:
It would be better and less error-prone to put these numbers in a constant

Reply:
Doesn't matter but I guess it's more readable, will do that 👍

Reply:
Doesn't matter but I guess it's more readable, will do that 👍

Reply:
I mean to have all the new codes with a special meaning in a constant dictionary, just to make it clear what they mean, without having to look into README